django/utils/text.py

```diff
@@ -87,10 +87,17 @@ class TruncationCompleted(Exception):
         "wbr",
     }
 
-    def __init__(self, *, convert_charrefs=True):
+    def __init__(self, *, num, truncate, convert_charrefs=True):
         super().__init__(convert_charrefs=convert_charrefs)
         self.tags = deque()
         self.output = ""
+        self.remaining = num
+        self.truncate = truncate
+
+    def handle_startendtag(self, tag, attrs):
+        self.handle_starttag(tag, attrs)
+        if tag not in self.void_elements:
+            self.handle_endtag(tag)
```

Comment:
This allows us to avoid unnecessary calls to `self.handle_endtag()` only to do nothing.